backend/app/services/adapters/task_kinds/queries.py

```
@@ -315,8 +803,275 @@ def get_task_detail(
+        bot_objects = kindReader.get_by_ids(db, KindType.BOT, list(all_bot_ids))
+
+        if not bot_objects:
+            return bots
+
+        # Collect all unique model and shell references for batch queries
+        model_refs = []
+        shell_refs = []
+        bot_id_to_refs = {}
+
+        for bot in bot_objects:
+            bot_crd = Bot.model_validate(bot.json)
```

Ruff (0.15.5)

[error] Undefined name `Bot`

(F821)